src/hypertable.c

```diff
@@ -2397,38 +2463,52 @@ ts_hypertable_has_compression_table(const Hypertable *ht)
  * 1. RowShareLock to SELECT for UPDATE
  * 2. UPDATE status using RowExclusiveLock
  */
-static bool
-hypertable_update_status_osm(Hypertable *ht)
+bool
+ts_hypertable_update_status_osm(Hypertable *ht)
 {
-	bool success = false;
-	ScanTupLock scantuplock = {
-		.waitpolicy = LockWaitBlock,
-		.lockmode = LockTupleExclusive,
-	};
-	ScanIterator iterator = ts_scan_iterator_create(HYPERTABLE, RowShareLock, CurrentMemoryContext);
-	iterator.ctx.index = catalog_get_index(ts_catalog_get(), HYPERTABLE, HYPERTABLE_ID_INDEX);
-	iterator.ctx.tuplock = &scantuplock;
-	ts_scan_iterator_scan_key_init(&iterator,
-								   Anum_hypertable_pkey_idx_id,
-								   BTEqualStrategyNumber,
-								   F_INT4EQ,
-								   Int32GetDatum(ht->fd.id));
+	FormData_hypertable form;
+	ItemPointerData tid;
+	/* lock the tuple entry in the catalog table */
+	bool found = lock_hypertable_tuple(ht->fd.id, &tid, &form);
+	Ensure(found, "hypertable id %d not found", ht->fd.id);
 
-	ts_scanner_foreach(&iterator)
+	int old_status = form.status;
+	if (old_status != ht->fd.status)
 	{
-		TupleInfo *ti = ts_scan_iterator_tuple_info(&iterator);
-		int status;
-		bool status_isnull;
+		form.status = ht->fd.status;
+		hypertable_update_catalog_tuple(&tid, &form);
+	}
+	return true;
+}
 
-		status = DatumGetInt32(slot_getattr(ti->slot, Anum_hypertable_status, &status_isnull));
-		Assert(!status_isnull);
-		if (status != ht->fd.status)
-		{
-			success = ts_hypertable_update(ht); // get RowExclusiveLock and update here
-		}
+bool
+ts_hypertable_update_chunk_sizing(Hypertable *ht)
+{
+	FormData_hypertable form;
+	ItemPointerData tid;
+	/* lock the tuple entry in the catalog table */
+	bool found = lock_hypertable_tuple(ht->fd.id, &tid, &form);
+	Ensure(found, "hypertable id %d not found", ht->fd.id);
```

Comment:
Is this message correct?

Reply:
Yes, locking failures have their error messages. So if we get false here, it means that we couldn't find the tuple.